File: backend/schemas/solicitud_schema.py
# backend/schemas/solicitud_schema.py
from marshmallow_sqlalchemy import SQLAlchemyAutoSchema
from marshmallow import fields
from models.solicitud import Solicitud
from extensions import db

class SolicitudSchema(SQLAlchemyAutoSchema):
    # Declara explícitamente ambos campos como required
    empresa = fields.String(
        required=True,
        error_messages={'required': 'La empresa es obligatoria'}
    )
    cargo = fields.String(
        required=True,
        error_messages={'required': 'El cargo es obligatorio'}
    )

    class Meta:
        model = Solicitud
        load_instance = True
        sqla_session = db.session

File: frontend/src/validation/solicitud.schema.js
// frontend/src/validation/solicitud.schema.js
import * as Yup from "yup";

export const SolicitudSchema = Yup.object().shape({
  empresa: Yup.string()
    .trim()
    .required("La empresa es obligatoria"),
  cargo: Yup.string()
    .trim()
    .required("El cargo es obligatorio"),
});
